homeassistant/components/enasolar/sensor.py

```diff
@@ -0,0 +1,248 @@
+
+        if enasolar.sun_up <= datetime.now().time() <= enasolar.sun_down:
+            values = await enasolar.read_data()
+        else:
+            values = False
+
+        for sensor in hass_data_sensors:
+            state_unknown = False
+            if not values and (
+                (sensor.sensor.per_day_basis and date.today() > sensor.sensor.date)
+                or (
+                    not sensor.sensor.per_day_basis
+                    and not sensor.sensor.per_total_basis
+                )
+            ):
+                state_unknown = True
+            sensor.async_update_values(unknown_state=state_unknown)
+            _LOGGER.debug(
+                "Data Sensor %s updated => %s", sensor.sensor.key, sensor.native_value
+            )
+        return values
+
+    def start_update_interval():
+        """Start the update interval scheduling."""
+        config_entry.async_on_unload(
+            async_track_time_interval_backoff(
+                hass, async_enasolar_meters, SCAN_METERS_MIN_INTERVAL
+            )
+        )
+        config_entry.async_on_unload(
+            async_track_time_interval_backoff(
+                hass, async_enasolar_data, SCAN_DATA_MIN_INTERVAL
+            )
+        )
+
+    if hass.is_running:
+        start_update_interval()
+    else:
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_update_interval)
+
+
+@callback
+def async_track_time_interval_backoff(hass, action, min_interval) -> CALLBACK_TYPE:
+    """Add a listener that fires repetitively and increases the interval when failed."""
+    remove = None
+    interval = min_interval
+
+    async def interval_listener():
+        """Handle elapsed interval with backoff."""
+        nonlocal interval, remove
+
+        try:
+            if await action():
+                interval = min_interval
+            else:
+                interval = min(interval * 2, SCAN_MAX_INTERVAL)
+        finally:
+            remove = async_call_later(hass, interval, interval_listener)
+
+    hass.async_create_task(interval_listener())
+
+    def remove_listener():
+        """Remove interval listener."""
+        remove()  # pylint: disable=not-callable
+
+    return remove_listener
+
+
+class EnaSolarSensor(SensorEntity):
+    """Representation of a EnaSolar sensor."""
+
+    def __init__(self, pyenasolar_sensor, inverter_name=None, serial_no=None):
+        """Initialize the EnaSolar sensor."""
+        self.sensor = pyenasolar_sensor
+        self._inverter_name = inverter_name
+        self.serial_no = serial_no
+        self._native_value = self.sensor.value
+
+        if pyenasolar_sensor.is_meter:
+            self._attr_state_class = STATE_CLASS_MEASUREMENT
+        else:
+            self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        self._attr_native_unit_of_measurement = ENASOLAR_UNIT_MAPPINGS[self.sensor.unit]
+        self._attr_should_poll = False
+
+    @property
+    def name(self):
```

Comment:
Use _attr_name * instead.
Goes also for the others below

Reply:
Sorry, don't understand this comment.

Reply:
I think thats the preferred method of setting attributes these days.

https://github.com/home-assistant/core/blob/caaa1b32c1808e0fd23f9ab42e1f0b481110883d/homeassistant/helpers/entity.py#L301-L308. https://github.com/home-assistant/core/blob/caaa1b32c1808e0fd23f9ab42e1f0b481110883d/homeassistant/components/sensor/__init__.py#L251-L263.